domain/modelmanager/state/state.go

```diff
@@ -46,9 +48,12 @@ func (s *State) Create(ctx context.Context, uuid model.UUID) error {
 func Create(ctx context.Context, uuid model.UUID, tx *sql.Tx) error {
 	stmt := "INSERT INTO model_list (uuid) VALUES (?);"
 	result, err := tx.ExecContext(ctx, stmt, uuid)
-	if err != nil {
+	if jujudb.IsErrConstraintPrimaryKey(err) {
+		return fmt.Errorf("model for uuid %q %w", uuid, modelerrors.AlreadyExists)
```

Comment:
We haven't discussed domain-sourced errors yet, but I'm thinking they make sense.

Reply:
I wouldn't say this work is the perfect case of it yet. I am struggling with the balance.

But what I am trying to achieve in creating a model is there can be a lot of touch points for NotFound. The generic error is far too big of a funnel for what wasn't found. Trying it to narrow down what wasn't found for the consumer.